api/src/controllers/notifications-controller.ts

```diff
@@ -0,0 +1,125 @@
+import { isNil } from "lodash"
+
+import logger from "@/utils/logger"
+
+import { Notification } from "@/models"
+import { NotificationsPolicy } from "@/policies"
+import { CreateService, UpdateService } from "@/services/notifications"
+import BaseController from "@/controllers/base-controller"
+
+export class NotificationsController extends BaseController<Notification> {
+  async index() {
+    try {
+      const where = this.buildWhere()
+      const scopes = this.buildFilterScopes()
+      const scopedNotifications = NotificationsPolicy.applyScope(scopes, this.currentUser)
+
+      const totalCount = await scopedNotifications.count({ where })
+      const notifications = await scopedNotifications.findAll({
+        where,
+        limit: this.pagination.limit,
+        offset: this.pagination.offset,
+        order: [["createdAt", "DESC"]],
+      })
+      return this.response.json({
+        notifications,
+        totalCount,
+      })
+    } catch (error) {
+      logger.error("Error fetching notifications" + error)
```

Comment:
This pattern of error logging doesn't work well in practice, since a lot of serious errors don't have a good "to string" function.

 I usually go with
```suggestion
      logger.error(`Error fetching notifications: ${error}`, { error })
```